core/src/test/java/com/linecorp/armeria/server/http/auth/AuthServiceTest.java

```diff
@@ -0,0 +1,129 @@
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.http.AbstractHttpService;
+import com.linecorp.armeria.server.logging.LoggingService;
+
+public class AuthServiceTest {
+    private static final Encoder BASE_64_ENCODER = Base64.getEncoder();
+    private static final Server server;
+    private static final Predicate<Entry<String, String>> basicAuthPredicate =
+            new Predicate<Entry<String, String>>() {
+                private Map<String, Integer> counter = Maps.newHashMap();
+
+                @Override
+                public boolean test(Entry<String, String> entry) {
+                    String username = entry.getKey();
+                    String password = entry.getValue();
+
+                    if (!username.equals(password)) {
+                        return false;
+                    }
+
+                    if (counter.containsKey(username)) {
+                        int count = counter.get(username);
+
+                        if (count >= username.length()) {
+                            return false;
+                        } else {
+                            counter.put(username, count + 1);
+                            return true;
+                        }
+                    } else {
+                        counter.put(username, 1);
+                        return true;
+                    }
+                }
+            };
+
+    private static int httpPort;
+
+    static {
+        final ServerBuilder sb = new ServerBuilder();
+
+        try {
+            sb.serviceAt(
+                    "/",
+                    new AbstractHttpService() {
+                        @Override
+                        protected void doGet(
+                                ServiceRequestContext ctx, HttpRequest req, HttpResponseWriter res) {
+                            res.respond(HttpStatus.OK);
+                        }
+                    }.decorate(HttpAuthService.basic(basicAuthPredicate))
+                     .decorate(LoggingService::new));
+        } catch (Exception e) {
+            throw new Error(e);
+        }
+        server = sb.build();
+    }
+
+    @BeforeClass
+    public static void init() throws Exception {
+        server.start().get();
+
+        httpPort = server.activePorts().values().stream()
+                         .filter(p -> p.protocol() == SessionProtocol.HTTP).findAny().get().localAddress()
+                         .getPort();
+    }
+
+    @AfterClass
+    public static void destroy() throws Exception {
+        server.stop();
+    }
+
+    @Test
+    public void testBasicAuth() throws Exception {
+        try (CloseableHttpClient hc = HttpClients.createMinimal()) {
+            try (CloseableHttpResponse res = hc.execute(basicAuthRequest("/", "a", "a"))) {
+                assertThat(res.getStatusLine().toString(), is("HTTP/1.1 200 OK"));
```

Comment:
Please also add the case where a client does not send a request with:

- no authorization header
- authorization header with different mechanism such as `Bearer`